internal/app/storage/database/database.go

```diff
@@ -70,29 +74,33 @@ func (s *DBStore) GetByOriginalURL(originalURL string) (string, error) {
 	return result, nil
 }
 
-func (s *DBStore) SaveWithConflict(shortURL, originalURL string) (string, error) {
+func (s *DBStore) GetAllByUser(userID string) ([]dto.UserURL, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
 
-	shortURL, err := s.queries.InsertOrGetShortURL(ctx, queries.InsertOrGetShortURLParams{
-		ShortUrl:    shortURL,
-		OriginalUrl: originalURL,
-	})
-	if err == nil {
-		return shortURL, nil
+	urls, err := s.queries.GetAllByUserID(ctx, sql.NullString{String: userID, Valid: true})
+	if err != nil {
+		return nil, err
 	}
 
-	// Проверяем тип ошибки
-	var pgErr *pgconn.PgError
-	if errors.As(err, &pgErr) && pgErr.Code == pgerrcode.UniqueViolation {
-		existingShortURL, err2 := s.GetByOriginalURL(originalURL)
-		if err2 != nil {
-			return "", err2
-		}
-		return existingShortURL, nil
+	result := make([]dto.UserURL, 0, len(urls))
+	for _, u := range urls {
+		result = append(result, dto.UserURL{
+			ShortURL:    u.ShortUrl,
+			OriginalURL: u.OriginalUrl,
+		})
 	}
+	return result, nil
+}
 
-	return "", err
+func (s *DBStore) BatchDelete(userID string, shortURLs []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
We need to pass the context from the handler, since the request duration in seconds is important to us.